config/initializers/devise.rb

```diff
@@ -0,0 +1,43 @@
+# Use this hook to configure devise mailer, warden hooks and so forth.
+# Many of these configuration options can be set straight in your model.
+Devise.setup do |config|
+  config.mailer_sender = 'please-change-me-at-config-initializers-devise@example.com'
+
+  require 'devise/orm/active_record'
+
+  config.case_insensitive_keys = [:email]
+
+  config.strip_whitespace_keys = [:email]
+
+  config.skip_session_storage = [:http_auth]
+
+  config.stretches = Rails.env.test? ? 1 : 11
+
+  config.reconfirmable = true
+
+  config.expire_all_remember_me_on_sign_out = true
+
+  config.password_length = 6..128
+
+  config.email_regexp = /\A[^@\s]+@[^@\s]+\z/
+
+  config.reset_password_within = 6.hours
+
+  config.sign_out_via = :delete
+
+  config.omniauth :saml,
+                  :assertion_consumer_service_url     => "http://localhost:3003/users/auth/saml/callback",
+                  :assertion_consumer_service_binding => "urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST",
+                  :issuer                             => Rails.configuration.x.saml_issuer,
+                  :idp_sso_target_url                 => Rails.configuration.x.idp_url,
+                  :idp_cert                           => File.read("#{Rails.root}/config/saml_cert.crt"),
+                  :name_identifier_format             => "urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress",
+                  :authn_context                      => "http://idmanagement.gov/ns/assurance/loa/1",
+                  :allowed_clock_drift                => 60,
+                  :certificate                        => File.read('config/demo_sp.crt'),
+                  :private_key                        => File.read('config/demo_sp.key'),
+                  :security                           => {:authn_requests_signed => true,
```

Comment:
This is most likely because I don't have the correct `demo_sp.key` that you are using, but to get the SP app to successfully redirect to the IdP app, I had to replace the `demo_sp.crt` with the one from the IdP app, add a `demo_sp.key` based on the `saml.key.enc` from the IdP app, and replace the `private_key` setting with this: `OpenSSL::PKey::RSA.new(File.read(Rails.root + 'config/demo_sp.key'), 'trust-but-verify').to_pem`

I'm pretty sure this should be resolved by providing the correct `demo_sp.key` in this PR since I think you said you were able to get the authentication working locally.